test/unit/test_manifest_selectors.py

```diff
@@ -0,0 +1,89 @@
+import dbt.exceptions
+import textwrap
+import yaml
+import unittest
+from dbt.config.selectors import SelectorDict
+
+
+def get_selector_dict(txt: str) -> dict:
+    txt = textwrap.dedent(txt)
+    dct = yaml.safe_load(txt)
+    return dct
+
+
+class SelectorUnitTest(unittest.TestCase):
+
+    def test_compare_cli_non_cli(self):
+        dct = get_selector_dict('''\
+            selectors:
+              - name: nightly_diet_snowplow
+                description: "This uses more CLI-style syntax"
+                definition:
+                  union:
+                    - intersection:
+                        - '@source:snowplow'
+                        - 'tag:nightly'
+                    - 'models/export'
+                    - exclude:
+                        - intersection:
+                            - 'package:snowplow'
+                            - 'config.materialized:incremental'
+                        - export_performance_timing
+              - name: nightly_diet_snowplow_full
+                description: "This is a fuller YAML specification"
+                definition:
+                  union:
+                    - intersection:
+                        - method: source
+                          value: snowplow
+                          childrens_parents: true
+                        - method: tag
+                          value: nightly
+                    - method: path
+                      value: models/export
+                    - exclude:
+                        - intersection:
+                            - method: package
+                              value: snowplow
+                            - method: config.materialized
+                              value: incremental
+                        - method: fqn
+                          value: export_performance_timing
+            ''')
+
+        sel_dict = SelectorDict.parse_from_selectors_list(dct['selectors'])
+        assert(sel_dict)
+        with_strings = sel_dict['nightly_diet_snowplow']['definition']
+        no_strings = sel_dict['nightly_diet_snowplow_full']['definition']
+        self.assertEqual(with_strings, no_strings)
```

Comment:
nice tests! do you think we should test every selection method for coverage? (e.g. `exposure`, `state`, `test_name`, and `test_type` methods)

Reply:
There isn't any code that handles the different selection methods differently, so I don't think it would be useful. This code doesn't do anything to validate (that's handled by the cli.py code) so pretty much anything ought to work, valid or not.

Reply:
I've added some tests for exposure. I did need to fix an issue with the '+' operator.